### PR TITLE
Security bump

### PR DESCRIPTION
**Pull request by kevinhughes27**
this bumps actionpack and deals with all the related upgrade shenanigans.

Main Changes
-------------------
* Use minitest
* had to upgrade to ActionDispatch and the api changed a bit (especially around accessing the request data)
* test assertion api changes

for reivew
@bizla @fw42

### CODE REVIEW

**Comment by fw42** [comment]
gem's shouldn't have a Gemfile.lock file, the dependencies and their versions should be in the gemspec (and only there)

File: oauth.gemspec [hunk 1]
```diff
@@ -23,11 +23,12 @@ Gem::Specification.new do |spec|
...
 
   spec.add_development_dependency(%q<rake>)
+  spec.add_development_dependency(%q<minitest>)
   spec.add_development_dependency(%q<byebug>)
   spec.add_development_dependency(%q<jeweler>)
-  spec.add_development_dependency(%q<actionpack>, [">= 2.3.8"])
+  spec.add_development_dependency(%q<actionpack>, [">= 4.0.2"])
   spec.add_development_dependency(%q<iconv>, ["~> 1.0.3"])
-  spec.add_development_dependency(%q<rack>, ["~> 1.1.0"])
+  spec.add_development_dependency(%q<rack>)
```
Comment (fw42) [inline oauth.gemspec]: should we `>=` this to a non-vulnerable version?

File: test/test_action_controller_request_proxy.rb [hunk 2]
```diff
@@ -1,13 +1,13 @@
 require File.expand_path('../test_helper', __FILE__)
 
 require 'oauth/request_proxy/action_controller_request'
-require 'action_controller/test_process'
+#require 'action_controller/test_process'
```
Comment (fw42) [inline test/test_action_controller_request_proxy.rb]: what's this? remove?

**Comment by fw42** [comment]
LGTM :+1:

File: test/test_action_controller_request_proxy.rb (hunk 2 above)
Comment (kevinhughes27) [inline test/test_action_controller_request_proxy.rb]: derp forgot I had commented this, it can be deleted

**Comment by fw42** [comment]
btw do we really need this fork? Can we try to get the changes merged upstream and then go back to that?

**Comment by kevinhughes27** [comment]
@fw42 there is a PR - no one has responded to it. I've pinged a couple of times

**Comment by kevinhughes27** [comment]
removed the Gemfile.lock and fixed a few more things to keep the tests running

**Comment by bizla** [comment]
:pizza:

**Comment by kevinhughes27** [comment]
@bizla no one bumped shopify core to use our fork. The accounting app is using this version but core is still on the latest from rubygems. Can you take this or delegate to someone on payments still maybe @ivanfer

**Comment by bizla** [comment]
@ivanfer seems beautifully suited to the task. 
Also, re: the PR being merged upstream, it looks like the preferred method of communication is through their Google group, according to the README. Did you happen to fire off an e-mail there?

**Comment by kevinhughes27** [comment]
I saw that too. I'll do so but looks like other people haven't had success either

**Pull request merged**